linux/LinuxProcessTable.c

```diff
@@ -1098,6 +1098,33 @@ static void LinuxProcessTable_readCwd(LinuxProcess* process, openat_arg_t procFd
    free_and_xStrdup(&process->super.procCwd, pathBuffer);
 }
 
+/*
+ * Read /proc/<pid>/syscall (thread-specific data)
+ */
+static void LinuxProcessTable_readSyscall(LinuxProcess* process, openat_arg_t procFd) {
+   char buffer[PATH_MAX];
+   ssize_t r;
+
+   r = xReadfileat(procFd, "syscall", buffer, sizeof(buffer));
+   if (r <= 0) {
+      free(process->syscall);
+      process->syscall = NULL;
+      return;
+   }
+
+   /* Remove '\n' */
+   buffer[r - 1] = '\0';
+
+   char* arg = strchr(buffer, ' ');
```

Comment:
Using `strchrnul` would allow to skip the following check. There's a proper implementation in `XUtils.h` you can use.

Reply:
If you think the kernel may remove `'\n'` in the future, and also for avoid duplicate scanning, I think use `strcspn` is more elegant.